openwebstart/src/main/java/com/openwebstart/jvm/RemoteRuntimeManager.java

```diff
@@ -44,6 +45,16 @@ private RemoteRuntimeManager() {
 
         LOG.debug("Trying to find remote Java runtime. Requested version: '{}' Requested vendor: '{}' requested os: '{}'", versionString, vendor, operationSystem);
 
+        // jvm server specified in jnlp is not null and is allowed
+        // TODO : What to do if not found in Whitelist
+        // TODO : confirm that we only check the URL in JNLP against whitelist and not the URL specified in settings
+        if (RuntimeManagerConfig.isNonDefaultServerAllowed() && specificServerEndpoint != null) {
+            if (!UrlWhiteListUtils.isUrlInWhitelist(specificServerEndpoint,  RuntimeManagerConfig.getJvmServerWhitelist())) {
+                LOG.warn("*************** JVM Server URL {} not in JVM Server Whitelist", specificServerEndpoint);
+                return Optional.empty();
```

Comment:
I ask myself if this is the right way to go. maybe throwing an exception or showing an error dialog makes more sense.

Reply:
This is what I want to discuss